dali/pipeline/basic/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Get all the source files and dump test files
+file(GLOB tmp *.cc *.cu)
+set(DALI_SRCS ${DALI_SRCS} ${tmp})
+file(GLOB tmp *_test.cc)
+remove(DALI_SRCS "${DALI_SRCS}" ${tmp})
```

Comment:
Just a small note here, cmake discourages using `file` for this use case:
> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.

Reply:
It's not like our approach is completely wrong, it got some advantages as well. For example it is terse compared to the other option. AFAIK if adding new files with glob approach forces you to call CMake by hand or to touch the CMakeLists.txt, you basically have to do as well when you list the files explicitly (by modifying the list of files). 

If we want to change our current approach, I think it should be done as _separate_ PR for whole project at once. I do not want to add that to this PR nor mix the approaches throughout the project.